official/utils/arg_parsers/parsers.py

```diff
@@ -58,9 +58,38 @@ def __init__(self):
 from __future__ import division
 from __future__ import print_function
 
-
 import argparse
 
+import tensorflow as tf
+
+
+# Map string to (TensorFlow dtype, default loss scale)
+DTYPE_MAP = {
+    "fp16": (tf.float16, 128),
+    "fp32": (tf.float32, 1),
+}
+
+
+def parse_dtype_info(flags):
+  """Convert dtype string to tf dtype, and set loss_scale default as needed.
+
+  Args:
+    flags: namespace object returned by arg parser.
+
+  Raises:
+    ValueError: If an invalid dtype is provided.
+  """
+  if not (flags.dtype is None or isinstance(flags.dtype, str)):
+    return  # Make function idempotent
```

Comment:
Shouldn't this instead be if dtype is in the set of allowed tf dtypes? Otherwise, you could pass in an int and get through, right? Although maybe that's being too defensive if we assume this only gets called with flags generated by the argparser, which requires a string.